src/apps/routes/apps/elements.clj

```diff
@@ -2,79 +2,55 @@
-        contain multiple types of information and some types of information can be described using
-        multiple data formats. For example, the Nexus format can contain multiple types of
-        information, including phylogenetic trees. And phylogenetic trees can also be represented in
-        PhyloXML format, and a large number of other formats. The file format and information type
-        together identify the type of input consumed by a tool or the type of output produced by a
-        tool."
+        :return elements-schema/InfoTypeListing
+        :summary elements-schema/AppElementsInfoTypeListingSummary
+        :description elements-schema/AppElementsInfoTypeListingDocs
         (ok (list-elements "info-types" params)))
 
   (GET "/parameter-types" []
         :query [params AppParameterTypeParams]
-        :return ParameterTypeListing
-        :summary "List App Parameter Types"
-        :description "Parameter types represent the types of information that can be passed to a tool. For
-        command-line tools, a parameter generally represents a command-line option and the parameter
-        type represents the type of data required by the command-line option. For example a
-        `Boolean` parameter generally corresponds to a single command-line flag that takes no
-        arguments. A `Text` parameter, on the other hand, generally represents some sort of textual
-        information. Some parameter types are not supported by all tool types, so it is helpful in
-        some cases to filter parameter types either by the tool type or optionally by the tool
-        (which is used to determine the tool type). If you filter by both tool type and tool ID then
-        the tool type will take precedence. Including either an undefined tool type or an undefined
-        tool type name will result in an error"
+        :return elements-schema/ParameterTypeListing
+        :summary elements-schema/AppElementsParameterTypeListingSummary
+        :description elements-schema/AppElementsParameterTypeListingDocs
         (ok (list-elements "parameter-types" params)))
 
   (GET "/rule-types" []
         :query [params SecuredQueryParams]
-        :return RuleTypeListing
-        :summary "List App Parameter Rule Types"
-        :description "Rule types represent types of validation rules that may be defined to validate user
-        input. For example, if a parameter value must be an integer between 1 and 10 then the
-        `IntRange` rule type may be used. Similarly, if a parameter value must contain data in a
-        specific format, such as a phone number, then the `Regex` rule type may be used."
+        :return elements-schema/RuleTypeListing
+        :summary elements-schema/AppElementsRuleTypeListingSummary
+        :description elements-schema/AppElementsRuleTypeListingDocs
         (ok (list-elements "rule-types" params)))
 
+  ;; Deprecated?
   (GET "/tools" []
```

Comment:
We can also mark this endpoint as `deprecated` in the Swagger UI, but the current version of `compojure-api` doesn't support the `deprecated` flag natively.

We could add support with a method like the following, perhaps in `common-swagger-api.routes`:
```clojure
(defmethod compojure.api.meta/restructure-param :deprecated
  [_ deprecated? acc]
  (update-in acc [:swagger] assoc :deprecated deprecated?))
```

Then adding `:deprecated true` to the route will mark it as deprecated in the Swagger UI: https://swagger.io/docs/specification/paths-and-operations/

Of course, if this endpoint is no longer in use, it's probably best to just remove it in this case, but I thought I'd take the opportunity to note the `deprecated` flag here.